### ISSUE TITLE
Investigate possibility of Buy Local (in-region) policies in EPS

### ISSUE THREAD

**Issue by robbieorvis** [Energy-Innovation/eps-us#81]
Explore possibility of EPS policy lever to adjust domestic/import shares in IO tables to estimate how buy domestic policies would affect jobs/GDP, etc...

**Comment by jrissman** [comment]
Does the buy local policy you want only affect how government spends any increases in revenue due to the policy package, such as new or increased carbon pricing?  Or do you want the buy local policy to be able to affect how government spends its BAU revenue?

Right now, it is trivial for me to do the former.  We don't currently have a representation of government BAU spending, so the latter would require new input data and structure.  It also would be odd for us to track BAU government spending but not BAU spending by households or industry, so we might want to start tracking all three using subscripts on the new structure.  We also would probably have to tackle the issue of deficit spending, which is not necessary today, as we are only looking at incremental changes due to policy.  So the request becomes a lot harder if you want the Buy Local policy to be able to affect BAU government spending, not just spending caused by the policy package.

A Buy Local policy that affects BAU spending isn't really a climate policy.  But a Buy Local policy that influences how revenues from climate policies are spent is more like a design feature of those policies than a policy in its own right.

On Jul 22, 2020 9:55 AM, Robbie Orvis <notifications@github.com> wrote:

Explore possibility of EPS policy lever to adjust domestic/import shares in IO tables to estimate how buy domestic policies would affect jobs/GDP, etc...

**Comment by ssonniaa** [comment]
If it is trivial to build the former, I would advocate strongly for us to do so. Would be great to be able to explore the impacts of this kind of policy design feature as we explore and test the jobs results for our policy packages and top policies.

**Comment by robbieorvis** [comment]
I was thinking the former, so support that.

________________________________
Robbie Orvis
Director of Energy Policy Design
Phone: 415-799-2171
98 Battery Street, Suite 202
San Francisco, CA  94111
www.energyinnovation.org<http://www.energyinnovation.org/>
[cid:image001.jpg@01D0D699.20A24470]
________________________________

Check out our new book, Designing Climate Solutions: A Policy Guide for Low-Carbon Energy<https://www.amazon.com/Designing-Climate-Solutions-Policy-Low-Carbon/dp/1610919564>
Available wherever books are sold

[Policy Design book cover]

**Comment by jrissman** [comment]
Okay.  One other thing to keep in mind.  We already have a policy lever that is used to represent government redirecting spending to specific ISIC codes, which represents using policy-driven revenue on things like infrastructure, or whatever else.  That policy already assumes those government revenue reallocations are to domestic suppliers.  So if you're using that policy to represent something like using carbon tax revenue to fund infrastructure construction, no new policy lever is needed.

A new policy lever could be added to affect government spending when it's _not_ being redirected to specific cash flow entities or ISIC codes using the government revenue redirection policy.  But our OECD data (in `io-model/GBbIC`) have 98% of government spending going to the ISIC codes for government services, defense, and education (ISIC 84 and ISIC 85).  It directs almost nothing to the various industry ISIC codes directly (though the Leontief Inverse Matrix causes some activity in those ISIC codes when activity in ISIC 84 is increased).  Looking at the domestic content share (in `io-model/DCSoCbIC`), government services is already 99.9% provided by domestic suppliers, and education is 98.6% from domestic suppliers.  So eliminating the foreign share of government spending won't do much on a direct level, given the OECD data we are using.

I think the Buy Local requirement ought to apply one level downstream also.  So insofar as ISIC 84 (government services and defense) buys things from industries, their purchases should also be 100% domestic when this policy is enabled.  I can approximate that by using the ISIC 84 row from `TLIM` instead of `DLIM`  in the domestic calculations (e.g. all the indirect and induced downstream activity globally becomes domestic activity).  I still wouldn't expect a huge effect (TLIM is only about 10% higher than DLIM for most cells in the ISIC 84 row), but at least this would properly capture a domestic-only requirement on spending by the entities within ISIC 84, not just the top-line government allocation to ISIC 84.

So in summary, I'm not sure this lever is going to make a huge impact, because _the government budget mostly goes to services that are already provided by domestic suppliers, including by the government itself_.  The Buy Local policies may have more impact when applied to specific government projects that involve purchases from industry (such as infrastructure construction), and we already have a separate lever for that.

**Comment by robbieorvis** [comment]
Hmmm, okay this gives me pause. I agree with the need to go downstream, BUT the issue is that we don’t want ALL of an industry’s change in output to be from domestic, only the share that comes from government. That starts to get pretty tricky to calculate… Perhaps we should table this for now.

________________________________
Robbie Orvis
Director of Energy Policy Design
Phone: 415-799-2171
98 Battery Street, Suite 202
San Francisco, CA  94111
www.energyinnovation.org<http://www.energyinnovation.org/>
[cid:image001.jpg@01D0D699.20A24470]
________________________________

Check out our new book, Designing Climate Solutions: A Policy Guide for Low-Carbon Energy<https://www.amazon.com/Designing-Climate-Solutions-Policy-Low-Carbon/dp/1610919564>
Available wherever books are sold

[Policy Design book cover]

**Comment by jrissman** [comment]
Maybe I didn't explain very well.  To handle downstream, I was only thinking about taking the row for ISIC 84 from `TLIM` while leaving all other rows from `DLIM` - that is, only dollars spent on ISIC 84 are respent entirely domestically, not dollars spent on other ISIC codes.  It does overstate the effect slightly because a dollar respent by ISIC 84 on, say, steel makers, then could be respent on foreign things by the steel makers.  But it wouldn't be in this case.  That's why I said I could "approximate" it by using the row for ISIC 84 from TLIM.  But that's a tertiary effect and vanishingly small.  I think it's a very good approximation.

**Comment by jrissman** [comment]
The issue I was flagging was really that the lever probably won't do much, not that the lever will over-state the effects.

**Comment by robbieorvis** [comment]
Okay. Well I’d be interested in testing it out and exploring the effects if you think it is methodologically sound.

________________________________
Robbie Orvis
Director of Energy Policy Design
Phone: 415-799-2171
98 Battery Street, Suite 202
San Francisco, CA  94111
www.energyinnovation.org<http://www.energyinnovation.org/>
[cid:image001.jpg@01D0D699.20A24470]
________________________________

Check out our new book, Designing Climate Solutions: A Policy Guide for Low-Carbon Energy<https://www.amazon.com/Designing-Climate-Solutions-Policy-Low-Carbon/dp/1610919564>
Available wherever books are sold

[Policy Design book cover]

**Comment by ssonniaa** [comment]
Just to help me understand, what happens with a clean energy standard, for example? In that policy, the government sets a target share of zero carbon resources and that stimulates a mix of government and private spending to build the needed infrastructure to meet the standard. So, in that case, a buy local provision could apply to private investment too, right?

**Comment by robbieorvis** [comment]
Or to give another example, there is proposed legislation right now that would promote ZEV sales and mandate a share (let’s assume 100% for simplicity) of the sales come from domestic manufacturers. Is that something that could be modeled?

________________________________
Robbie Orvis
Director of Energy Policy Design
Phone: 415-799-2171
98 Battery Street, Suite 202
San Francisco, CA  94111
www.energyinnovation.org<http://www.energyinnovation.org/>
[cid:image001.jpg@01D0D699.20A24470]
________________________________

Check out our new book, Designing Climate Solutions: A Policy Guide for Low-Carbon Energy<https://www.amazon.com/Designing-Climate-Solutions-Policy-Low-Carbon/dp/1610919564>
Available wherever books are sold

[Policy Design book cover]

**Comment by jrissman** [comment]
Oh, I see.  I didn't realize that was what you were thinking of for a buy local policy.  I thought it was just confined to how the government respent money the government itself gained from the policy package, like a carbon tax.  A Buy local policy that is binding on private entities isn't something I was thinking about when writing any of my comments above.

I suppose any entity can be modeled as spending any policy-driven change in revenue domestically by treating the domestic content share of that entity's purchases as 1.  In a way, it's easier than making government spending local because we don't run into the ISIC 84 issue.

The challenge would be that the government can't mandate that all spending by private entities be domestic, as it could for its own spending.  I think these policies would only affect a small share of the production of any given ISIC code.  For instance, many LDVs are sold in the U.S., so shifting a few sales from foreign car makers to domestic car makers would be small in percentage terms.  So we couldn't just use a foreign content share of zero for that ISIC code.  We'd need to reduce the foreign content share by some percentage.  This percentage would have to take account of the fact that each ISIC code produces more types of stuff than the covered products.  For example, the same ISIC code that makes EVs also makes diesel trucks, which would not be affected by that policy.  We have total output by ISIC code for a historical year but no future year time series, though we do calculate a rough future year series ourselves, but one we currently only rely on for very limited outputs (primarily the "change in average compensation per employee" graph).

There is also a question of additionality.  If a policy mandates that at least Y% of EV sales (amounting to X vehicles) must come from domestic manufacturers, but currently 50X cars per year are made and sold by domestic manufacturers, it might be easy for entities affected by the policy to ensure the covered X vehicles are produced domestically but reduce sales of other domestic vehicles by X, so you still end up selling 50X domestically-produced vehicles, not 51X.

Here's one thing I can do that would give you this policy lever but make the complexity tractable.  I add a policy lever that lets you reduce the foreign content share by a user-specified percentage, subscripted by ISIC code.  If you want it to affect the upstream purchases of that ISIC code also, we could scale between DLIM and TLIM for that same ISIC code by the same percentage, or a percentage specified in a second policy lever.  (The difference here is whether the policy says the cars have to be made by a domestic manufacturer, or whether the domestic manufacturer has to buy domestically-produced inputs like steel, or both.)  Then it's up to you to translate any given Buy Local policy into a meaningful setting for this lever for any affected ISIC codes.  You could use the BAU output from each ISIC code to help you estimate the denominator.  You'd need to figure out the numerator yourself based on the properties of the policy you're trying to model, like how many cars are affected, what is their average purchase price, and are they all additional or are some offset by substitution of foreign cars elsewhere.

Would giving you a lever (or two) to reduce the foreign content share of particular ISIC codes, and/or to scale between DLIM and TLIM for particular ISIC codes, be satisfactory?

If anything here is unclear, maybe we could chat on the phone.

**Comment by jrissman** [comment]
I see this Buy Local policy is flagged for inclusion 3.3.  I can do this on my next EPS workday, but _I need you to answer one quick question for me_, so I know which structural approach to use.

The Buy Local policy will be subscripted by Industry Category.  Suppose you decide to enable the policy at some non-zero setting for a particular industry, such as Road Vehicle Manufacturing.  In terms of the policy this represents, do you want the model to interpret the policy to be requiring:

A) Purchasers of road vehicles (such as consumers, taxi companies, etc.) purchase more road vehicles made in the modeled region.

B) Road vehicle manufacturers purchase more steel, plastic, glass, and other input materials and components (which go into making the road vehicles) made in the modeled region.

C) Both

Once I know which of these designs you intend for this "Buy Local" policy, I can program it accordingly.

**Comment by robbieorvis** [comment]
Hi Jeff,

I think it is both, but perhaps as independent options…

For instance, one use case is where the government institutes a Buy America place in which additional government purchase (say of electric vehicles) must be from American companies.

There is another use case where we model domestic-manufacturing incentive policies as increasing the domestic content share of commodities for certain industries. This came up in the some 2035 transportation report we worked on with LBL, where there was a desire to model increase purchase of battery equipment from domestic manufacturers as an input to the auto manufacturing industry.

So I think they are both useful and actually perhaps distinct policy levers.

What do you think?

-R

________________________________
Robbie Orvis
Director of Energy Policy Design
Phone: 415-799-2171
98 Battery Street, Suite 202
San Francisco, CA  94111
www.energyinnovation.org<http://www.energyinnovation.org/>
***@***.***D7466F.1B4945E0]
________________________________

Check out our book, Designing Climate Solutions: A Policy Guide for Low-Carbon Energy<https://www.amazon.com/Designing-Climate-Solutions-Policy-Low-Carbon/dp/1610919564>
Available wherever books are sold

[Policy Design book cover]

**Comment by robbieorvis** [comment]
Hi Jeff,

Thanks for this. How do you feel about “In-Region” instead of “Local”? Sometimes “Buy Local” implies to me the food chain and buying local produce, where as “In-Region” makes it clear it’s in the model region?

Yes, understood that buying in-region inputs is limited to industries. That makes sense. On buying outputs, two thoughts:


  1.  Remember that we aren’t just programming this for the United States, and in some other regions, the government purchases a much higher share of things. For example in India, the government basically owns 100% of the bus fleet (I am not sure what the default domestic content share is, but wouldn’t be surprised if many buses are imported from China). In the EU, I think there is a large government procurement program for vehicles as well. Does that change your mind about whether to subscript by entity?
  2.  One of the benefits of having things broken out by entity is actually to highlight the precise point you are making in the US: buying American is often touted as a way the government can make a dent in emissions but the emissions impact is tiny… to that end having the ability to show that in the model is kind of useful as a way to suggest that other policies are needed.

Do either of those points affect your decision?


________________________________
Robbie Orvis
Director of Energy Policy Design
Phone: 415-799-2171
98 Battery Street, Suite 202
San Francisco, CA  94111
www.energyinnovation.org<http://www.energyinnovation.org/>
***@***.***D74680.0BECB760]
________________________________

Check out our book, Designing Climate Solutions: A Policy Guide for Low-Carbon Energy<https://www.amazon.com/Designing-Climate-Solutions-Policy-Low-Carbon/dp/1610919564>
Available wherever books are sold

[Policy Design book cover]

**Comment by robbieorvis** [comment]
That all makes sense and you are right about the concern with the narrow scope of government-only purchases, excluding other projects government hires out for. With that in mind, I’m okay with not getting into that level detail. For what it’s worth, though, I wasn’t considering limiting it to just government purchases but instead was thinking of subscripting the policy lever by purchasing entity, to give users the flexibility the test different options.

________________________________
Robbie Orvis
Director of Energy Policy Design
Phone: 415-799-2171
98 Battery Street, Suite 202
San Francisco, CA  94111
www.energyinnovation.org<http://www.energyinnovation.org/>
***@***.***D74683.ED6665D0]
________________________________

Check out our book, Designing Climate Solutions: A Policy Guide for Low-Carbon Energy<https://www.amazon.com/Designing-Climate-Solutions-Policy-Low-Carbon/dp/1610919564>
Available wherever books are sold

[Policy Design book cover]

**Comment by jrissman** [comment]
This morning I went ahead and tested every configuration of the "Buy In-Region Outputs" policy (i.e. disaggregating the effects by which buyer is affected, which is the proposed second subscript dimension).  All Industry Categories are affected with a 100% setting in all years.  (There is no need to phase in the Buy In-Region Outputs policy because it merely says how the money is directed, it doesn't create cash flows, so it already is "phased in" as the policies that create the changes in cash flows phase in.)  The results are shown in the graph below and full calculations and data appear in the attached spreadsheet file.

![BuyInRegionChangeInGDP](https://user-images.githubusercontent.com/7120106/118020148-9d9a9600-b30e-11eb-917b-49503a46bb7d.PNG)

Spreadsheet:
[2021-05-12 Test of Buy In-Region Affected Buyers.xlsx](https://github.com/Energy-Innovation/eps-us/files/6468091/2021-05-12.Test.of.Buy.In-Region.Affected.Buyers.xlsx)

We can't even see the blue wedge, so it is clear that the IO data we have are not _directly_ ascribing purchases from the Industry Categories to government to any meaningful degree, so it would be inappropriate to use our IO data to disaggregate the effects of the Buy In-Region Outputs policy by purchasing entity.  Therefore, these tests confirm my initial suspicions, and I won't be subscripting this policy by purchasing entity.  It will be subscripted by Industry Category.

If you want to model a policy that directs only the government to buy in-region products, you'll need to use an outside source to identify the percentage of the outputs of a particular industry that are purchased by (or whose purchase is funded by) government and set the Buy In-Region Outputs lever accordingly.

Note also that applying the policy to **all** buyers has some interactive effects that make the total (black line) greater than the sum of the parts (green and orange wedges).  This is normal.

**Comment by jrissman** [comment]
Also, note that the Buy In-Region Outputs and Buy In-Region Inputs policies will only be affecting the changes in expenditures caused by the policy package, not the BAU expenditures by all actors in society, which are vastly larger and which we do not currently run through the IO model.  In other words, the Buy In-Region policies in the EPS are aimed at modeling what happens if you direct to in-region product the decarbonization-related spending (and associated spending, such as respending of fuel savings on other goods/services) by all actors in society.  It does not model the shifting of BAU spending patterns that have no relationship to any energy or climate policy, which we don't model in detail anyhow.

**Comment by robbieorvis** [comment]
Great, thanks!

________________________________
Robbie Orvis
Director of Energy Policy Design
Phone: 415-799-2171
98 Battery Street, Suite 202
San Francisco, CA  94111
www.energyinnovation.org<http://www.energyinnovation.org/>
***@***.***D7473E.3173DC20]
________________________________

Check out our book, Designing Climate Solutions: A Policy Guide for Low-Carbon Energy<https://www.amazon.com/Designing-Climate-Solutions-Policy-Low-Carbon/dp/1610919564>
Available wherever books are sold

[Policy Design book cover]

**Comment by jrissman** [comment]
This is complete, though it could use some testing.

Remember that these policies only affect the changes in cash flows caused by the policy package.  They don't change BAU cash flows to industries, which we don't track explicitly.

The Buy In-Region Outputs lever generally improves jobs and GDP, especially when applied to industries that do not consume a lot of fossil fuels in order to make their products, such as computer products or electrical equipment and appliances.  It reduces jobs and GDP if you apply it to an industry that is receiving large negative cash flows from the rest of the policy package (such as coal producers, refined petroleum product producers, and road vehicle producers), because it causes domestic businesses to absorb a larger share of those negative cash flows.  If the effect seems odd to you (why would domestic car makers suffer if some buyers of foreign cars instead buy domestic cars?), remember that the Buy In-Region policy only applies to the change in cash flow caused by the policy package, not the industry's total sales, and the NDC policy package is causing car-makers to have small negative cash flows for some reason unrelated to the new policies.  While I'm sure car-makers would love to have a Buy In-Region requirement that applied to all sales of vehicles, that's not what this policy is doing - it's only being applied to the policy-driven cash flow changes, which are negative.

I worry that this type of interaction has the potential to be very confusing to model users because this policy doesn't necessarily work the way they expect when the policy package causes a negative cash flow to an industry like road vehicle manufacturing.  If this is too confusing, we may feel we need to modify the policy to apply to an industry's total revenue, not just the policy-driven change in the revenue, which would require us to use somewhat rough estimates of future revenue divided up by ISIC code.  (We have good projections of Value Added / GDP going forward, but those projections aren't divided up by ISIC code.  All our ISIC code data is historical in nature - no projections of how the relative allocation to different ISIC codes changes in the future,. Still, it might be worth doing if the lever's effects are too counterintuitive the way it is currently programmed.

The Buy In-Region Inputs lever works similarly to the Buy In-Region Outputs lever, but it affects the suppliers of the industry you pick rather than affecting only the industry you pick.  But keep in mind that most industries are also significant suppliers to themselves, muddying the waters somewhat.  I haven't quite looked into why some industries have positive or negative effects on overall GDP and Jobs when using this lever, but I think it comes down again to the way the lever only currently applies to the change in cash flow due to the policy package, which is often negative for specific industries.  That is, it's likely the same reason as we encountered with the simpler Buy In-Region Outputs policy.

I'm not going to close this issue yet because I suspect we might decide we need to make these policies apply to the total cash flow of each industry and not just to the cash flow changes caused by the policy package to get more easily-understood and recognizable results.

The latest model with the new policies in place is on the staging server now.  Let me know what you think.

**Comment by jrissman** [comment]
You know, maybe the buy in-region policies are fine as is.  They just serve to accentuate when the policy package already causes positive or negative impacts on a particular industry, by funneling all those impacts to domestic businesses.  Maybe that's not too hard to explain to users.

But it is really counter-intuitive that they hurt an industry like automakers until you explain the policy's limited scope.

I'm not sure how the policy's effects are going to be interpreted by someone with fresh eyes.  It could really use testing and opinions from the rest of the EPS team, not just whether it is working in a basic sense, but more importantly, whether it provides the analysis capabilities you need and that a user would expect from Buy In-Region policies.

**Comment by robbieorvis** [comment]
Thanks for this, Jeff. I will take a look next week. I suspect you may be right regarding the need to move this to total spending, but without trying it I’m not sure. I’m just trying to relate it back to how it is being discussed and the issues surrounding how it will be perceived, as you alluded to. I will take a look next week.

________________________________
Robbie Orvis
Director of Energy Policy Design
Phone: 415-799-2171
98 Battery Street, Suite 202
San Francisco, CA  94111
www.energyinnovation.org<http://www.energyinnovation.org/>
***@***.***D748D0.E6787390]
________________________________

Check out our book, Designing Climate Solutions: A Policy Guide for Low-Carbon Energy<https://www.amazon.com/Designing-Climate-Solutions-Policy-Low-Carbon/dp/1610919564>
Available wherever books are sold

[Policy Design book cover]

**Comment by jrissman** [comment]
Thank you for taking a look next week!

Reminder to self: We need to consolidate the two Buy In-Region policy levers into one.  The way to represent the existing "Buy In-Region Inputs" policy as part of a single unified policy is to apply the DLIM -> TLIM shift selectively to policy-targeted industries where they serve as suppliers for other industries (not as the industry being supplied).  This will keep the treatment of respending on an industry's products consistent with the treatment of direct spending on that industry's products.  As is, those sources effectively have with different domestic content shares if the two policy levers are set to different values, which should not be possible for a single industry.

I think it is a more accurate (and also more user-friendly) design that I wasn't able to think of until I had built a working Buy In-Region Inputs policy.

**Comment by jrissman** [comment]
In commit 3447b00, I went ahead and caused the "Buy In-Region Outputs" lever to govern all mechanisms, as I now think is required for correct simulation of a Buy In-Region policy.  I didn't remove the Buy In-Region Inputs policy yet (I'm waiting on your testing before I do this), but that lever doesn't do anything anymore.  I will remove it after you test the Buy In-Region Outputs feature.

As a reminder, we can't split these into two levers because the inputs of one industry are the outputs of another, so a "Buy In-Region Outputs" lever isn't having the intended effect if it ignores these types of purchases, which are very important for certain industries (such as steelmakers) that largely sell business-to-business.  The current implementation is more correct.  It also is simpler because it only involves one lever, and that lever acts as most people would assume.  (E.g. when applied to the steel industry, it affects purchases of steel).

**Comment by jrissman** [comment]
Tested and verified working on stage.

The policy now can shift BAU imports to domestic suppliers, not just shift the policy-induced changes in imports.  This produces results that are much more in-line with intuition about what the policy means and in-line with how such a policy would really be designed.

Note that at 100% setting (shifting all imports to domestic production), unsurprisingly, the policy has pretty large effects.  The fact that we have energy service demand feedback loops already implemented is important here, because we are able to model the increase in domestic emissions that comes from the increase in domestic production caused by this policy.

Lastly, I wanted to highlight for the team this note, which I added to the guidance text for this policy, since I think it is important for end users:

> **Implementation Note:** This policy reduces imports. In reality, other regions might retaliate by restricting their own purchases of goods produced by the modeled region. However, the EPS only considers policies within the modeled region, so the Buy In-Region Products policy does not affect export levels (i.e., other countries/regions are assumed not to retaliate).

I recommend not using high settings of this lever, as lower settings would represent policies that would be less likely to spark a trade war / retaliatory tariffs or restrictions.  (I don't think we can realistically model a trade war, which would play out according to a range of political, economic, and diplomatic factors that we can't predict from our raw data.)

I set the max bound on this policy in WebAppData to 50%, but you might wish to adjust it to some other value.
